### PR TITLE
[CSS] chore: GH issues 18758 contain property should mention 'inline-size'

### LINKED ISSUE #18758: Mention inline-size containment
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/contain

### What specific section or headline is this issue about?

Values

### What information was incorrect, unhelpful, or incomplete?

The page does not mention the `inline-size` value. 

### What did you expect to see?

I expected to see the `inline-size` value listed as a possible value for the `contain` CSS property.

### Do you have any supporting links, references, or citations?

This is documented in the [CSS Containment Level 3 specification](https://www.w3.org/TR/css-contain-3/#inline-size); it is implemented in WebKit, Firefox, and Chromium.

- [WebKit bug 236354](https://bugs.webkit.org/show_bug.cgi?id=236354)
- [Mozilla bug 1755565](https://bugzilla.mozilla.org/show_bug.cgi?id=1755565)

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/contain`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/contain
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/contain/index.md
* Last commit: https://github.com/mdn/content/commit/4e7553185bcaf7f30ec7fec9ae35aa3e8c787bfa
* Document last modified: 2022-07-11T21:06:17.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
Fixes https://github.com/mdn/content/issues/18758

This PR adds some details for `inline-size` value for the `contain` property.

__Notes:__
Opening as draft as I would like to clarify:
- [x] strict containment is equivalent to `size layout paint`

#### Related issues

- https://github.com/mdn/content/issues/18758

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]
Looks great to me, with one suggestion

File: files/en-us/web/css/contain/index.md [hunk 1]
```diff
@@ -66,11 +70,13 @@ The `contain` property is specified as either one of the following:
 - `content`
   - : Indicates that all containment rules except `size` are applied to the element. This is equivalent to `contain: layout paint style`.
 - `size`
-  - : Indicates that the element can be sized without the need to examine its descendants' sizes.
+  - : Indicates that size containment is applied to the element. The size of the element can be computed in isolation, ignoring the child elements. Cannot be combined with `inline-size`.
+- `inline-size`
+  - : Indicates that inline size containment is applied to the element. The inline size of the element can be computed in isolation, ignoring the child elements. Cannot be combined with `size`.
 - `layout`
-  - : Indicates that nothing outside the element may affect its internal layout and vice versa.
+  - : Indicates that the internal layout of the element is isolated from the rest of the page.
```
Comment (teoli2003) [inline files/en-us/web/css/contain/index.md:77]: (Overall looks good). I'm not sure that _isolated_ is well-known. What about:
```suggestion
  - : Indicates that the internal layout of the element is isolated from the rest of the page, that is, nothing outside the element affects its internal layout, and vice versa.
```
That way the term _isolated_ is introduced to the reader, with a definition.

File: files/en-us/web/css/contain/index.md [hunk 2]
```diff
@@ -66,11 +70,13 @@ The `contain` property is specified as either one of the following:
 - `content`
   - : Indicates that all containment rules except `size` are applied to the element. This is equivalent to `contain: layout paint style`.
 - `size`
-  - : Indicates that the element can be sized without the need to examine its descendants' sizes.
+  - : Indicates that size containment is applied to the element. The size of the element can be computed in isolation, ignoring the child elements. Cannot be combined with `inline-size`.
```
Comment (dipikabh) [inline files/en-us/web/css/contain/index.md:73]:
```suggestion
  - : Indicates that size containment is applied to the element. The size of the element can be computed in isolation, ignoring the child elements. This value cannot be combined with `inline-size`.
```

File: files/en-us/web/css/contain/index.md [hunk 3]
```diff
@@ -66,11 +70,13 @@ The `contain` property is specified as either one of the following:
...
+- `inline-size`
+  - : Indicates that inline size containment is applied to the element. The inline size of the element can be computed in isolation, ignoring the child elements. Cannot be combined with `size`.
```
Comment (dipikabh) [inline files/en-us/web/css/contain/index.md:75]:
```suggestion
  - : Indicates that inline size containment is applied to the element. The inline size of the element can be computed in isolation, ignoring the child elements. This value cannot be combined with `size`.
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
Tiny nits

File: files/en-us/web/css/contain/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/css/contain/index.md]: very good, thanks 👍🏻

**Review by caugner — COMMENTED** [review]
Looks like this PR has two unrelated changes:
1. Updating the `contain` page.
2. Adding `CSS Containment 3` to `SpecData.json`.

Is this intended?

**Comment by bsmth** [comment]
Hi @caugner thanks for checking. 
>     1. Updating the `contain` page.
The clarifications on the opening paragraph help with overall context on the purpose of the contain property, in my opinion. Does this look okay to you?

>     2. Adding `CSS Containment 3` to `SpecData.json`.

I assumed this needed attention, but this is [no longer maintained](https://github.com/mdn/content/blob/main/files/en-us/mdn/writing_guidelines/howto/json_structured_data/index.md#specdata-specification-information). I have reverted in https://github.com/mdn/content/pull/19736/commits/7577e7140c0ae1d7e8e3c216f85cfa266eef4b1a

**Comment by caugner** [comment]
> Does this look okay to you?

I was only looking at the PR, because changes to certain files (including SpecData.json) add `@mdn/core-dev` as a reviewer. I'll leave it to the others to review the content change. 🙂 

> I assumed this needed attention, but this is [no longer maintained](https://github.com/mdn/content/blob/main/files/en-us/mdn/writing_guidelines/howto/json_structured_data/index.md#specdata-specification-information). I have reverted in [7577e71](https://github.com/mdn/content/commit/7577e7140c0ae1d7e8e3c216f85cfa266eef4b1a)

TIL that SpecData.json is deprecated (I didn't know). 😄

**Review by dipikabh — APPROVED** [review]
LGTM! Thanks @bsmth  🎉

**Comment by bsmth** [comment]
Thanks, all 🎉

**Pull request merged**